### PR TITLE
fix: allow checkbox group within radio group

### PR DESCRIPTION
**Pull request by vinteo**
## Description

This allows the `CheckboxGroup` of variant `default` to be used within a `RadioGroup` with variant `box` without the CSS for the box `RadioGroup` affecting the `CheckboxGroup` items.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by pphminions — DISMISSED** [review]
LGTM 🐈‍⬛

**Review by romi-h — DISMISSED** [review]

File: src/components/Checkbox/index.jsx [hunk 1]
```diff
@@ -74,8 +74,9 @@ const Checkbox = ({
       onKeyDown={onCheckboxKeyDown}
       {...expandDts(dts)}
     >
-      <label htmlFor={id}>
+      <label className="aui--checkbox-label-container" htmlFor={id}>
```
Comment (xiaofan2406) [inline src/components/Checkbox/index.jsx:77]: can we shorten this to just `aui--checkbox-label`

File: src/components/Checkbox/styles.css [hunk 2]
```diff
@@ -7,19 +7,19 @@ $label-gap: 8px;
   cursor: pointer;
   display: block;
 
-  & label {
+  & label.aui--checkbox-label-container {
```
Comment (xiaofan2406) [inline src/components/Checkbox/styles.css:10]: does it work without `label` now? reducing level would make overriding easier

File: src/components/Checkbox/styles.css [hunk 3]
```diff
@@ -7,19 +7,19 @@ $label-gap: 8px;
...
     cursor: pointer;
     font-weight: $font-weight-normal;
     margin-bottom: 0;
     display: flex;
   }
 
-  & input {
+  & input.aui--checkbox-input {
```
Comment (xiaofan2406) [inline src/components/Checkbox/styles.css:17]: same here, if they work without the tags, we can use the classname only

File: src/components/Checkbox/index.jsx (hunk 1 above)
Comment (renrizzolo) [reply src/components/Checkbox/index.jsx:77]: `aui--checkbox-label` is used to wrap the label text within the label.
I think we should swap the class names so the label element has the label class and the text wrapper has the container class.

```jsx
<label className="aui--checkbox-label">
  <div className="aui--checkbox-input-container">
    ...
  <div className="aui--checkbox-label-container">
   ...
```

Comment (vinteo) [reply src/components/Checkbox/index.jsx:77]: there is already an `aui--checkbox-label`

Comment (vinteo) [reply src/components/Checkbox/index.jsx:77]: it looks like there are some overrides in direct web for that class

**Review by xiaofan2406 — APPROVED** [review]

**Review by renrizzolo — APPROVED** [review]

**Comment by pphminions** [comment]
merging 🚀
